indexing-service/src/main/java/io/druid/indexing/common/task/ConvertSegmentTask.java

```diff
@@ -342,9 +341,13 @@ public String getType()
     @Override
     public TaskStatus run(TaskToolbox toolbox) throws Exception
     {
-      log.info("Subs are good!  Italian BMT and Meatball are probably my favorite.");
-      convertSegment(toolbox, segment, indexSpec, force, validate);
-      return success();
+      if(toolbox.getTaskActionClient().submit(new LockUpgradeAction(getInterval()))) {
+        log.info("Subs are good!  Italian BMT and Meatball are probably my favorite.");
+        convertSegment(toolbox, segment, indexSpec, force, validate);
```

Comment:
this is actually not its own task; the upgrade should be hoisted up to the real task. (it just runs a loop over these things)

And because of that, this is also a lot of work to be doing in a non-preemptible way, and maybe should be chunked (see comment on the archive task).